### ISSUE TITLE
Miband does not fetch activity data nor shows sleep monitor

### ISSUE THREAD

**Issue by xphnx** [Freeyourgadget/Gadgetbridge#77]
Here are my reports:
- The App Manager is empty
- The sleep monitor is empty too. The first day shows a the graph of a lapse of 2h aprox, but it was steatic, and after a disconnection of the device it gets empty.
- Battery level, FW and debug works great

Logs [ActivityDatabase](https://my.owndrive.com/public.php?service=files&t=388c6cf84a31569039f69d3d6f3d01a6)
[ActivityDatabase-journal](https://my.owndrive.com/public.php?service=files&t=9a028cad6a0052d8517997660fc3e89d)

FW: 1.0.9.3
Smartphone with CM 12.1
Gadgetbridge v0.4.0 (but I built from source 0.4.1, and seems to be the same)

Love.

**Comment by cpfeiffer** [comment]
Thanks for your report!

1) App Manager is Pebble only -- we need to change our activity navigation and logic to avoid such "gaps".
I'll have a look at the database. Can you explain what actually happend? You invoked "Fetch activity data" and then "Sleep Monitor" and then it only displayed data of 2 hours?

The sleep monitor works even in disconnected mode, so I don't undertand what you mean that it get's empty when disconnected.

I'm on Firmware 1.0.4.3.

**Comment by xphnx** [comment]
Well, now I remember better: the first day I have the band I invoked Fetch Activity data and then Sleep monitor, and it shows a graph displaying the data of 2 hours. Then I see that the graph didnt update, and I built 0.4.1 from source. This version shows the Sleep monitor empty. 

I came back to the F-droid version 0.4.0 waiting for the update and the Sleep monitor was empty yet.

**Comment by cpfeiffer** [comment]
At the moment, the Sleep Monitor shows activities of the last 24 hours. You currently have to manually fetch data again to get updated data. Your old data simply went "out of scope". It's still there in the db for later.

**Comment by xphnx** [comment]
I always invoke Fetch Data before go to Sleep Monitor. 

Screenshot of the Sleep Monitor attached.
![screenshot_2015-06-21-22-20-57](https://cloud.githubusercontent.com/assets/4024412/8273242/791fb120-1864-11e5-9f43-df7c317cc343.png)

PD: Maybe it would be nice the data was fetching automatically from time to time, but it's just and idea.

**Comment by cpfeiffer** [comment]
Yeah, either that, or move the "Fetch" action into the Sleep activity. I'm wondering if this is firmware related. Do you have anything special in the logfile (haven't looked into the db yet).

**Comment by xphnx** [comment]
A part of the log, if it can help:
" 06-21 22:39:44.353 W/PackageManager(778): Failure retrieving resources for nodomain.freeyourgadget.gadgetbridge: Resource ID #0x0
06-21 22:41:40.669 D/nodomain.freeyourgadget.gadgetbridge.btle.BtLEQueue(13742): characteristic changed: 0000ff03-0000-1000-8000-00805f9b34fb
06-21 22:42:41.148 I/Timeline(1255): Timeline: Activity_launch_request id:nodomain.freeyourgadget.gadgetbridge time:81918111
06-21 22:42:41.149 I/ActivityManager(587): START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=nodomain.freeyourgadget.gadgetbridge/.ControlCenter (has extras)} from uid 10019 on display 0
06-21 22:42:41.153 V/WindowManager(587): addAppToken: AppWindowToken{2daa5324 token=Token{28007fb7 ActivityRecord{272606b6 u0 nodomain.freeyourgadget.gadgetbridge/.ControlCenter t2985}}} to stack=1 task=2985 at 0
06-21 22:42:41.174 V/WindowManager(587): Adding window Window{3092e2bc u0 Starting nodomain.freeyourgadget.gadgetbridge} at 9 of 15 (after Window{3c31d36f u0 com.nolanlawson.logcat/com.nolanlawson.logcat.LogcatActivity})
06-21 22:42:41.279 V/WindowManager(587): Adding window Window{399b8966 u0 nodomain.freeyourgadget.gadgetbridge/nodomain.freeyourgadget.gadgetbridge.ControlCenter} at 9 of 16 (before Window{3092e2bc u0 Starting nodomain.freeyourgadget.gadgetbridge})
06-21 22:42:41.287 D/nodomain.freeyourgadget.gadgetbridge.BluetoothCommunicationService(13742): Service startcommand: nodomain.freeyourgadget.gadgetbride.bluetoothcommunicationservice.action.start
06-21 22:42:41.287 D/nodomain.freeyourgadget.gadgetbridge.BluetoothCommunicationService(13742): Service startcommand: nodomain.freeyourgadget.gadgetbride.bluetoothcommunicationservice.action.request_versioninfo
06-21 22:42:41.336 I/nodomain.freeyourgadget.gadgetbridge.GB(13742): Setting broadcast receivers to: true
06-21 22:42:41.374 I/ActivityManager(587): Displayed nodomain.freeyourgadget.gadgetbridge/.ControlCenter: +175ms
06-21 22:42:41.695 I/Timeline(587): Timeline: Activity_windows_visible id: ActivityRecord{272606b6 u0 nodomain.freeyourgadget.gadgetbridge/.ControlCenter t2985} time:81918658
06-21 22:42:42.813 V/WindowManager(587): Adding window Window{11850443 u0 AtchDlg:nodomain.freeyourgadget.gadgetbridge/nodomain.freeyourgadget.gadgetbridge.ControlCenter} at 10 of 16 (after Window{399b8966 u0 nodomain.freeyourgadget.gadgetbridge/nodomain.freeyourgadget.gadgetbridge.ControlCenter})
06-21 22:42:43.707 D/nodomain.freeyourgadget.gadgetbridge.BluetoothCommunicationService(13742): Service startcommand: nodomain.freeyourgadget.gadgetbride.bluetoothcommunicationservice.action.fetch_activity_data
06-21 22:42:43.707 D/nodomain.freeyourgadget.gadgetbridge.btle.BtLEQueue(13742): about to add: Transaction task: fetch activity data with 2 actions
06-21 22:42:43.712 I/nodomain.freeyourgadget.gadgetbridge.GB(13742): Setting broadcast receivers to: true
06-21 22:42:44.786 D/nodomain.freeyourgadget.gadgetbridge.btle.BtLEQueue(13742): characteristic write: 0000ff05-0000-1000-8000-00805f9b34fb (success)
06-21 22:42:44.787 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): handleControlPoint got status:0
06-21 22:42:44.787 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): handleControlPoint GOT DATA:0x 6
06-21 22:42:44.787 D/nodomain.freeyourgadget.gadgetbridge.btle.BtLEQueue(13742): characteristic changed: 0000ff07-0000-1000-8000-00805f9b34fb
06-21 22:42:44.787 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): total data to read: 0 len: 0 minute(s)
06-21 22:42:44.787 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): data to read until next header: 0 len: 0 minute(s)
06-21 22:42:44.788 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): TIMESTAMP: 19/6/2015 23:03:23 magic byte: 0
06-21 22:42:44.788 D/nodomain.freeyourgadget.gadgetbridge.btle.BtLEQueue(13742): about to add: Transaction task: send acknowledge with 1 actions
06-21 22:42:46.031 D/nodomain.freeyourgadget.gadgetbridge.btle.BtLEQueue(13742): characteristic write: 0000ff05-0000-1000-8000-00805f9b34fb (success)
06-21 22:42:46.032 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): handleControlPoint got status:0
06-21 22:42:46.032 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): handleControlPoint GOT DATA:0x a
06-21 22:42:46.032 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): handleControlPoint GOT DATA:0x f
06-21 22:42:46.032 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): handleControlPoint GOT DATA:0x 5
06-21 22:42:46.034 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): handleControlPoint GOT DATA:0x 13
06-21 22:42:46.034 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): handleControlPoint GOT DATA:0x 17
06-21 22:42:46.034 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): handleControlPoint GOT DATA:0x 3
06-21 22:42:46.035 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): handleControlPoint GOT DATA:0x 17
06-21 22:42:46.035 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): handleControlPoint GOT DATA:0x 0
06-21 22:42:46.035 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): handleControlPoint GOT DATA:0x 0
06-21 22:42:46.037 I/nodomain.freeyourgadget.gadgetbridge.GB(13742): Setting broadcast receivers to: true
06-21 22:42:47.239 V/WindowManager(587): Adding window Window{2987314a u0 AtchDlg:nodomain.freeyourgadget.gadgetbridge/nodomain.freeyourgadget.gadgetbridge.ControlCenter} at 10 of 16 (after Window{399b8966 u0 nodomain.freeyourgadget.gadgetbridge/nodomain.freeyourgadget.gadgetbridge.ControlCenter})
06-21 22:42:47.989 I/Timeline(13742): Timeline: Activity_launch_request id:nodomain.freeyourgadget.gadgetbridge time:81924952
06-21 22:42:47.990 I/ActivityManager(587): START u0 {cmp=nodomain.freeyourgadget.gadgetbridge/.activities.SleepChartActivity (has extras)} from uid 10114 on display 0
06-21 22:42:47.993 V/WindowManager(587): addAppToken: AppWindowToken{1cee5931 token=Token{34c9d1d8 ActivityRecord{8d0b8bb u0 nodomain.freeyourgadget.gadgetbridge/.activities.SleepChartActivity t2985}}} to stack=1 task=2985 at 1
06-21 22:42:48.053 I/nodomain.freeyourgadget.gadgetbridge.activities.SleepChartActivity(13742): number of samples:0
06-21 22:42:48.060 V/WindowManager(587): Adding window Window{26482597 u0 nodomain.freeyourgadget.gadgetbridge/nodomain.freeyourgadget.gadgetbridge.activities.SleepChartActivity} at 11 of 17 (after Window{2987314a u0 AtchDlg:nodomain.freeyourgadget.gadgetbridge/nodomain.freeyourgadget.gadgetbridge.ControlCenter EXITING})
06-21 22:42:48.137 I/ActivityManager(587): Displayed nodomain.freeyourgadget.gadgetbridge/.activities.SleepChartActivity: +118ms
06-21 22:42:48.519 I/Timeline(587): Timeline: Activity_windows_visible id: ActivityRecord{8d0b8bb u0 nodomain.freeyourgadget.gadgetbridge/.activities.SleepChartActivity t2985} time:81925481
06-21 22:42:49.098 W/PackageManager(778): Failure retrieving resources for nodomain.freeyourgadget.gadgetbridge: Resource ID #0x0
06-21 22:43:31.565 I/Timeline(1255): Timeline: Activity_launch_request id:nodomain.freeyourgadget.gadgetbridge time:81968528
06-21 22:43:31.566 I/ActivityManager(587): START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=nodomain.freeyourgadget.gadgetbridge/.ControlCenter (has extras)} from uid 10019 on display 0
06-21 22:43:32.176 I/Timeline(587): Timeline: Activity_windows_visible id: ActivityRecord{8d0b8bb u0 nodomain.freeyourgadget.gadgetbridge/.activities.SleepChartActivity t2985} time:81969139
06-21 22:43:33.549 I/Timeline(587): Timeline: Activity_windows_visible id: ActivityRecord{272606b6 u0 nodomain.freeyourgadget.gadgetbridge/.ControlCenter t2985} time:81970512
06-21 22:43:34.665 V/WindowManager(587): Adding window Window{b79d74c u0 AtchDlg:nodomain.freeyourgadget.gadgetbridge/nodomain.freeyourgadget.gadgetbridge.ControlCenter} at 10 of 16 (after Window{399b8966 u0 nodomain.freeyourgadget.gadgetbridge/nodomain.freeyourgadget.gadgetbridge.ControlCenter})
06-21 22:43:35.536 D/nodomain.freeyourgadget.gadgetbridge.BluetoothCommunicationService(13742): Service startcommand: nodomain.freeyourgadget.gadgetbride.bluetoothcommunicationservice.action.fetch_activity_data
06-21 22:43:35.536 D/nodomain.freeyourgadget.gadgetbridge.btle.BtLEQueue(13742): about to add: Transaction task: fetch activity data with 2 actions
06-21 22:43:35.540 I/nodomain.freeyourgadget.gadgetbridge.GB(13742): Setting broadcast receivers to: true
06-21 22:43:36.453 D/nodomain.freeyourgadget.gadgetbridge.btle.BtLEQueue(13742): characteristic write: 0000ff05-0000-1000-8000-00805f9b34fb (success)
06-21 22:43:36.454 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): handleControlPoint got status:0
06-21 22:43:36.454 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): handleControlPoint GOT DATA:0x 6
06-21 22:43:36.455 D/nodomain.freeyourgadget.gadgetbridge.btle.BtLEQueue(13742): characteristic changed: 0000ff07-0000-1000-8000-00805f9b34fb
06-21 22:43:36.455 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): total data to read: 0 len: 0 minute(s)
06-21 22:43:36.455 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): data to read until next header: 0 len: 0 minute(s)
06-21 22:43:36.456 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): TIMESTAMP: 19/6/2015 23:03:23 magic byte: 0
06-21 22:43:36.456 D/nodomain.freeyourgadget.gadgetbridge.btle.BtLEQueue(13742): about to add: Transaction task: send acknowledge with 1 actions
06-21 22:43:37.698 D/nodomain.freeyourgadget.gadgetbridge.btle.BtLEQueue(13742): characteristic write: 0000ff05-0000-1000-8000-00805f9b34fb (success)
06-21 22:43:37.698 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): handleControlPoint got status:0
06-21 22:43:37.699 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): handleControlPoint GOT DATA:0x a
06-21 22:43:37.699 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): handleControlPoint GOT DATA:0x f
06-21 22:43:37.699 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): handleControlPoint GOT DATA:0x 5
06-21 22:43:37.699 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): handleControlPoint GOT DATA:0x 13
06-21 22:43:37.699 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): handleControlPoint GOT DATA:0x 17
06-21 22:43:37.700 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): handleControlPoint GOT DATA:0x 3
06-21 22:43:37.700 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): handleControlPoint GOT DATA:0x 17
06-21 22:43:37.700 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): handleControlPoint GOT DATA:0x 0
06-21 22:43:37.700 I/nodomain.freeyourgadget.gadgetbridge.miband.MiBandSupport(13742): handleControlPoint GOT DATA:0x 0
06-21 22:43:37.700 I/nodomain.freeyourgadget.gadgetbridge.GB(13742): Setting broadcast receivers to: true
06-21 22:43:38.889 V/WindowManager(587): Adding window Window{f983b77 u0 AtchDlg:nodomain.freeyourgadget.gadgetbridge/nodomain.freeyourgadget.gadgetbridge.ControlCenter} at 10 of 16 (after Window{399b8966 u0 nodomain.freeyourgadget.gadgetbridge/nodomain.freeyourgadget.gadgetbridge.ControlCenter})
06-21 22:43:39.832 I/Timeline(13742): Timeline: Activity_launch_request id:nodomain.freeyourgadget.gadgetbridge time:81976795
06-21 22:43:39.833 I/ActivityManager(587): START u0 {cmp=nodomain.freeyourgadget.gadgetbridge/.activities.SleepChartActivity (has extras)} from uid 10114 on display 0
06-21 22:43:39.839 V/WindowManager(587): addAppToken: AppWindowToken{26c9702 token=Token{1c60104d ActivityRecord{3af9c3e4 u0 nodomain.freeyourgadget.gadgetbridge/.activities.SleepChartActivity t2985}}} to stack=1 task=2985 at 1
06-21 22:43:39.903 I/nodomain.freeyourgadget.gadgetbridge.activities.SleepChartActivity(13742): number of samples:0
06-21 22:43:39.908 V/WindowManager(587): Adding window Window{38813a50 u0 nodomain.freeyourgadget.gadgetbridge/nodomain.freeyourgadget.gadgetbridge.activities.SleepChartActivity} at 11 of 17 (after Window{f983b77 u0 AtchDlg:nodomain.freeyourgadget.gadgetbridge/nodomain.freeyourgadget.gadgetbridge.ControlCenter EXITING})
06-21 22:43:39.982 I/ActivityManager(587): Displayed nodomain.freeyourgadget.gadgetbridge/.activities.SleepChartActivity: +119ms
06-21 22:43:40.350 I/Timeline(587): Timeline: Activity_windows_visible id: ActivityRecord{3af9c3e4 u0 nodomain.freeyourgadget.gadgetbridge/.activities.SleepChartActivity t2985} time:81977313
06-21 22:43:40.897 W/PackageManager(778): Failure retrieving resources for nodomain.freeyourgadget.gadgetbridge: Resource ID #0x0 "

**Comment by danielegobbetti** [comment]
Hello @xphnx and thank you for the log (I formatted it a bit better by adding three backticks ( ` ) at the beginning and at the end)

According to the log the band seems to be "stuck" at the timestamp: 19/6/2015 23:03:23 without any further data to send. The transfer seems to have no issues but no data is detected.

Do you remember if you did something with the band (like a fw upgrade using the original app?) in that moment?

Could you try to reboot the band (from GadgetBridge -> Debug screen) and then try to transfer the data again (please keep an eye on the log!)

I have the same firmware version as you (1.0.9.3) hence we can rule out incompatibilities.

**Comment by xphnx** [comment]
I have never updated the firmware nor used the privative app. 
I rebooted de band, now it seem the date is right but the problem persists. GBridge is not able to fetch data.
I dont see nothing interesting in the log (but i am not an expert...)

**Comment by danielegobbetti** [comment]
Thanks for testing! 

When you say that the date is right do you mean that the TIMESTAMP line in logcat shows the current time?
--

**Comment by xphnx** [comment]
Is this ok?
![screenshot_2015-06-22-19-49-19](https://cloud.githubusercontent.com/assets/4024412/8288692/0859073e-1918-11e5-8ebc-e85b937ec81b.png)

**Comment by danielegobbetti** [comment]
Not really. I've never seen the band getting stuck like this. Normally each timestamp should appear only once with a positive value of elements to fetch.

Another test, could you try to use the "set time" button in the debug activity and see if it helps?
--

**Comment by xphnx** [comment]
Seems not to work. I think only the reboot makes a "Timestamp" appears.

After a reboot:
![screenshot_2015-06-22-19-56-04](https://cloud.githubusercontent.com/assets/4024412/8288844/e8631cca-1918-11e5-8721-0f2d6ce323be.png)

**Comment by danielegobbetti** [comment]
Mhh... Can you please check what you set in the miband settings (name, height, ...)? Perhaps some settings are needed to make the tracking work.

If everything looks right, could you try changing your name and restarting GB? That should trigger a new pairing and perhaps things will start working after it. (To accept the pairing you have to touch/move the band while it's blinking and vibrating)

--

**Comment by xphnx** [comment]
All the settings was set previously.
I change the profile name and as you said it goes OK. Now works. As a side effect the battery status is not showed, but its a minor problem for me.
I have noticed in other pairing processes the band doesnt blink and vibrates, only the first time (that match while the app was able to fetch data). So I think if the band doesn blink and vibrate the pairing is no completely ok, although notifications works.
Solved. 
Thank you very much  for your dedication Daniele.
